### PR TITLE
Fix ResponseCache.Copy{To,From} to properly update the value checksums.

### LINKED ISSUE #534: invalid checksum for key after merge-ranges
Starting on a newly created and empty cluster:

```
~ ./cockroach put a 1 b 2 c 3 d 4
~ ./cockroach split-range c c
~ ./cockroach scan
"a"	1
"b"	2
"c"	3
"d"	4
~ ./cockroach merge-range b
~ ./cockroach scan
I0401 09:44:51.147463    1360 kv.go:125] failed Scan: failed to send RPC: too many errors encountered (1 of 1 total): data.go:286: invalid checksum for key i	res-�����c�F�  �O, value bytes:"\"\020\n\016\022\014\010\370\315\310\367\355\367\271\350\023\020\000" checksum:1787442783
scan failed: failed to send RPC: too many errors encountered (1 of 1 total): data.go:286: invalid checksum for key i	res-�����c�F�  �O, value bytes:"\"\020\n\016\022\014\010\370\315\310\367\355\367\271\350\023\020\000" checksum:1787442783
```

### PR DESCRIPTION
**Pull request by petermattis**
Fixes #534.

### CODE REVIEW

File: storage/response_cache.go [hunk 1]
```diff
@@ -151,10 +151,23 @@ func (rc *ResponseCache) CopyInto(e engine.Engine, destRaftID int64) error {
...
 		if err != nil {
-			return false, util.Errorf("could not decode a response cache key %q: %s", kv.Key, err)
+			return false, util.Errorf("could not decode a response cache key %s: %s",
+				proto.Key(kv.Key), err)
 		}
-		encKey := engine.MVCCEncodeKey(engine.ResponseCacheKey(destRaftID, &cmdID))
-		return false, e.Put(encKey, kv.Value)
+		key := engine.ResponseCacheKey(destRaftID, &cmdID)
+		encKey := engine.MVCCEncodeKey(key)
+		// log.Infof("CopyInto: %s", encKey)
```
Comment (bdarnell) [inline storage/response_cache.go]: Remove this

**Comment by bdarnell** [comment]
LGTM

**Comment by BramGruneir** [comment]
The change looks good, but how about adding some unit tests?

File: storage/response_cache.go (hunk 1 above)
Comment (petermattis) [inline storage/response_cache.go]: Oops. Done.

**Comment by petermattis** [comment]
@BramGrunier the existing response cache `Copy{To,From}` tests fail with the change to mvccGetInternal to verify the checksums. The rest of this change fixes those tests.

File: storage/response_cache.go [hunk 2]
```diff
@@ -151,10 +151,21 @@ func (rc *ResponseCache) CopyInto(e engine.Engine, destRaftID int64) error {
 		// write it to the corresponding key in the new cache.
 		cmdID, err := rc.decodeResponseCacheKey(kv.Key)
 		if err != nil {
-			return false, util.Errorf("could not decode a response cache key %q: %s", kv.Key, err)
+			return false, util.Errorf("could not decode a response cache key %s: %s",
```
Comment (spencerkimball) [inline storage/response_cache.go]: I wonder whether this code shouldn't just use MVCCScan...? This has gotten ugly with the decoding of MVCCMetadata. Violated abstraction, etc.

**Comment by spencerkimball** [comment]
Curious if you reran the benchmarks for MVCCGet and MVCCScan to see how the addition of inline csum verification may have affected performance.

File: storage/response_cache.go (hunk 2 above)
Comment (petermattis) [inline storage/response_cache.go]: Yeah, I'll take a look at switching this to MVCCScan. Definitely ugly the way it is now.

**Comment by petermattis** [comment]
No, I didn't re-run the MVCC{Get,Scan} benchmarks. I'll do so tomorrow.

**Comment by petermattis** [comment]
I can't measure a speed difference between turning the checksum verification on or leaving it off. I can see the checksum calculation consumes 6% of the cpu, but the benchmark seems to be bottlenecked by GC.

**Pull request merged**